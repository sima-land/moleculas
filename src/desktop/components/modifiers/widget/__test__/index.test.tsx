import React from 'react';
import { act } from 'react-dom/test-utils';
import { mount } from 'enzyme';
import { ModifiersWidget } from '..';
import { Link } from '@sima-land/ui-nucleons/link';

describe('<ModifiersWidget />', () => {
  it('should renders correctly with prop', () => {
    const onSelectItem = jest.fn();
    const wrapper = mount(
      <ModifiersWidget
        title='test'
        items={[
          { content: 'Foo', selected: true },
          { content: 'Bar', selected: false },
          { content: 'Baz', selected: false },
        ]}
        sizesTableUrl='sima-land.ru'
        onSelectItem={onSelectItem}
      />
    );

    expect(wrapper.find('h3').prop('className')).toEqual('modifiers-title');
    expect(wrapper.find(Link).prop('className')).toEqual('modifiers-link');
    expect(wrapper).toMatchSnapshot();

    expect(onSelectItem).toBeCalledTimes(0);

    act(() => {
      wrapper.find('[data-testid="modifier-button"]').at(0).simulate('click');
    });
    wrapper.update();

    expect(onSelectItem).toBeCalledTimes(0);

    act(() => {
      wrapper.find('[data-testid="modifier-button"]').at(1).simulate('click');
    });
    wrapper.update();

    expect(onSelectItem).toBeCalledTimes(1);

    act(() => {
      wrapper.find('[data-testid="modifier-button"]').at(2).simulate('click');
    });
    wrapper.update();

    expect(onSelectItem).toBeCalledTimes(2);
  });
});
