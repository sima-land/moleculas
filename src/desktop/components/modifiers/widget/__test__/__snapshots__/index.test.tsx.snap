// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ModifiersWidget /> should renders correctly with prop 1`] = `
<ModifiersWidget
  items={
    Array [
      Object {
        "content": "Foo",
        "selected": true,
      },
      Object {
        "content": "Bar",
        "selected": false,
      },
      Object {
        "content": "Baz",
        "selected": false,
      },
    ]
  }
  onSelectItem={[MockFunction]}
  sizesTableUrl="sima-land.ru"
  title="test"
>
  <div>
    <h3
      className="modifiers-title"
    >
      test
       (
      3
       
      варианта
      ):
    </h3>
    <ModifiersGroup
      getItemContent={[Function]}
      getItemCount={[Function]}
      items={
        Array [
          Object {
            "content": "Foo",
            "selected": true,
          },
          Object {
            "content": "Bar",
            "selected": false,
          },
          Object {
            "content": "Baz",
            "selected": false,
          },
        ]
      }
      onSelectItem={[Function]}
    >
      <div
        className="modifiers-group height-limited"
        data-testid="data-tetsid"
      >
        <ForwardRef(ModifierButton)
          className="guttered"
          key="0"
          onClick={[Function]}
          selected={true}
        >
          <div
            className="modifier-button selected guttered"
            data-testid="modifier-button"
            onClick={[Function]}
          >
            <div
              className="content"
            >
              <span
                className="text guttered"
              />
            </div>
          </div>
        </ForwardRef(ModifierButton)>
        <ForwardRef(ModifierButton)
          className="guttered"
          key="1"
          onClick={[Function]}
          selected={false}
        >
          <div
            className="modifier-button guttered"
            data-testid="modifier-button"
            onClick={[Function]}
          >
            <div
              className="content"
            >
              <span
                className="text guttered"
              />
            </div>
          </div>
        </ForwardRef(ModifierButton)>
        <ForwardRef(ModifierButton)
          className="guttered"
          key="2"
          onClick={[Function]}
          selected={false}
        >
          <div
            className="modifier-button guttered"
            data-testid="modifier-button"
            onClick={[Function]}
          >
            <div
              className="content"
            >
              <span
                className="text guttered"
              />
            </div>
          </div>
        </ForwardRef(ModifierButton)>
        <ForwardRef(ModifierButton)
          className="show-all-button guttered invisible"
          content="+3"
          data-testid="modifier-group:show-all-button"
          onClick={[Function]}
          squared={true}
        >
          <div
            className="modifier-button squared show-all-button guttered invisible"
            data-testid="modifier-group:show-all-button"
            onClick={[Function]}
          >
            <div
              className="content"
            >
              <span
                className="text"
              >
                +3
              </span>
            </div>
          </div>
        </ForwardRef(ModifierButton)>
      </div>
    </ModifiersGroup>
    <ForwardRef(Link)
      className="modifiers-link"
      href="sima-land.ru"
    >
      <a
        className="link modifiers-link color__brand-blue hover-color__brand-deep-blue"
        data-testid="anchor"
        href="sima-land.ru"
      >
        Таблица размеров
      </a>
    </ForwardRef(Link)>
  </div>
</ModifiersWidget>
`;
